### DISCUSSION TITLE
New release / tag

### DISCUSSION THREAD

**Discussion by rdewell** [davdroman/PreciseDecimal#7] [General]
Hi @davdroman – looks like quite a few good changes have been made since the last release. Any chance you could tag a new version, or should we stick with "main" for the time being? Thanks for the very useful and necessary work!

**Comment by davdroman** [comment]
Hi @rdewell. Thank you for your question.

I've been trying to decide what to do with this library since it [doesn't fully address](https://github.com/davdroman/PreciseDecimal#is-this-solution-bullet-proof) all use cases properly.

The changes in main are quite disruptive. It moves away from relying on doubles, in order to embrace full-on string-only initialization and encoding/decoding. This is a step down from the current release, but I suspect it might be the safest way to go forward.

```swift
// latest release (1.1.0)
let foo: PreciseDecimal = 3.133 // valid, not guaranteed for 6+ decimal places
// main (~= 2.0.0)
let foo: PreciseDecimal = 3.133 // invalid
let foo: PreciseDecimal = "3.133" // valid, guaranteed for any amount of decimal places
```

This also makes `PreciseDecimal` an even thinner wrapper around `Decimal`, and one could argue this library becomes redundant in that case, but imo this is not the case. The reason being that `Decimal` still suffers from an inconvenient and minefield-y set of APIs, which cause an all too tempting inclination for users to forget about its limitations and "hold it wrong". So instead, `PreciseDecimal`'s advantage consists of a much more strict API to ensure safety while Apple bridges these annoying gaps in the `Decimal` type.

The only reason I haven't made a breaking release with these changes is doubt. So since you asked, I'd kindly like to know your thoughts on this to help me make a decision.

Does my rationale make sense to you?

**Reply by davdroman** [reply]
This idea has evolved over time. Here's the most up-to-date rationale on the whole thing: #10.
